File: Flyweight/FlyweightGame/FlyweightGame/SoldierType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FlyweightGame
{
    public class SoldierType
    {
        public string Type { get; }
        public string Color { get; }
        public string Animation { get; }

        public SoldierType(string type, string color, string animation)
        {
            Type = type;
            Color = color;
            Animation = animation;

            Console.WriteLine($"Creating type: {type}");
        }

        public void Display(int x, int y)
        {
            Console.WriteLine($"Displaying {Type} at ({x}, {y}) with color {Color} and animation {Animation}");
        }
    }
}

File: Flyweight/JavaExample/FlyweightGame/src/SoldierType.java
public class SoldierType {
    private final String type;
    private final String color;
    private final String animation;

    public SoldierType(String type, String color, String animation) {
        this.type = type;
        this.color = color;
        this.animation = animation;
        System.out.println("Creating type: " + type + " [" + color + "]");
    }

    public void display(int x, int y) {
        System.out.println("Displaying " + type + " at (" + x + ", " + y +
                ") with color " + color + " and animation " + animation);
    }
}
